ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/SendRequest.java

```diff
@@ -0,0 +1,46 @@
+package com.mozilla.telemetry.contextualservices;
+
+import com.mozilla.telemetry.ingestion.core.Constant.Attribute;
+import com.mozilla.telemetry.transforms.FailureMessage;
+import com.mozilla.telemetry.transforms.PubsubConstraints;
+import java.io.UncheckedIOException;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.WithFailures.ExceptionElement;
+import org.apache.beam.sdk.transforms.WithFailures.Result;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+/**
+ * Send GET requests to reporting.
+ */
+public class SendRequest extends
+    PTransform<PCollection<PubsubMessage>, Result<PCollection<PubsubMessage>, PubsubMessage>> {
+
+  public static SendRequest of() {
+    return new SendRequest();
+  }
+
+  @Override
+  public Result<PCollection<PubsubMessage>, PubsubMessage> expand(
+      PCollection<PubsubMessage> messages) {
+    return messages.apply(
+        MapElements.into(TypeDescriptor.of(PubsubMessage.class)).via((PubsubMessage message) -> {
+          message = PubsubConstraints.ensureNonNull(message);
+
+          // TODO: HTTP GET
+          System.out.println(message.getAttribute(Attribute.REPORTING_URL));
+
+          return message;
+        }).exceptionsInto(TypeDescriptor.of(PubsubMessage.class))
+            .exceptionsVia((ExceptionElement<PubsubMessage> ee) -> {
+              try {
+                throw ee.exception();
```

Comment:
I'm just starting the sending part now so it's still uncertain.  I was thinking of retrying a few times with exponential backoff and if it repeatedly fails we can write the failed messages to bigquery where we have a mechanism to manually retry from.

Do you have thoughts on the retry logic?  Like how many times to retry, time between retries, and which error codes should be retried.

Reply:
More specifically, I'm not sure if we should retry for 400's if we son't expect 400's in a valid request.  But 500's seem like they should be retried

Reply:
5xx codes should generally indicate a transient condition. I think it makes sense to include retry logic with exponential backoff for 5xx. Anything else that's not a 2xx can flow to errors.

But I'd also be happy to keep this super-simple for the initial draft and carve out 5xx handling to a future PR.

Reply:
The retry on 5xx errors in here sounds good to me. Would it be possible to log the request errors somewhere? It'll be useful to have that if that doesn't complicate the system for this release.

Reply:
> The retry on 5xx errors in here sounds good to me. Would it be possible to log the request errors somewhere? It'll be useful to have that if that doesn't complicate the system for this release.

That sounds like good motivation to avoid all retry logic for the initial release. If we let the 5xx messages flow to the error output, then we'll get the full error messages along with them, which could be helpful for making decisions about specific error handling logic we want to implement.

Reply:
Having the visual on error rates (even if the retry eventually succeeds) allows us to understand the overall health of the data sharing channel, also it can alert ourselves as well as the partner if the error happens too often.

To be clear, this could be done later. Let's have something deadly simple (such as retry those 5xx errors) for now.